src/binding/python/Attributable.cpp

```diff
@@ -104,12 +104,20 @@ bool setAttributeFromBufferInfo(
         // stride handling
         for( auto d = 0; d < buf.ndim; ++d )
         {
-            // std::cout << "    stride '" << d << "': " << buf.strides[d] << " - " << buf.shape[d] << std::endl;
-            if( buf.strides[d] != buf.shape[d] ) // general criterial
-                if( d > 0 && buf.strides[d] != 1 ) // ok for byte strings
+            // std::cout << "    stride '" << d << "': "
+            //           << buf.strides[d] / buf.itemsize
+            //           << " - " << buf.shape[d] << std::endl;
+            if( buf.strides[d] / buf.itemsize != buf.shape[d] ) // general criteria
```

Comment:
Under what condition is this branch not `true`?

```python
import numpy as np

import openPMD

DATA = np.arange(100).reshape((4, 25))

series = openPMD.Series('path', openPMD.Access_Type.create)
rc = series.iterations[1].particles['e']['position']['x']
rc.store([0, 0], [2, 25], DATA[::2, :])
series.flush()
```
?

Reply:
Yes, views for example but  you can also allocate non-contiguous memory in numpy arrays itself.